public/javascripts/application.js

```diff
@@ -147,5 +147,14 @@ $(function()
     {
         applicationNS.ask(message, { Yes: callback, No: null });
     };
+
+    applicationNS.confirmDestruction = function(callback)
+    {
+        if (odkmaker.data.clean === false)
+            odkmaker.application.confirm('Are you sure? You will lose unsaved changes to the current form.', callback);
+
```

Comment:
nit: i don't like this extra blank line, in case the else ends up wandering away.